plugin/plugin.go

```diff
@@ -423,6 +575,35 @@ func splitId(in, expectedService, expectedResource string) (string, string, erro
 	return splitId[3], splitId[7], nil
 }
 
+// concatenate the vmss name with the vmss instance id to get the external name
+func getExternalNameforVMSSInstance(in string) (string, error) {
+	splitId := strings.Split(strings.TrimLeft(in, "/"), "/")
+	if len(splitId) != 10 ||
+		!strings.EqualFold(splitId[0], constSubscriptions) ||
+		!strings.EqualFold(splitId[2], constResourceGroups) ||
+		!strings.EqualFold(splitId[4], constProviders) ||
+		!strings.EqualFold(splitId[5], constMsComputeService) ||
+		!strings.EqualFold(splitId[6], constVirtualMachineScaleSetsResource) ||
+		!strings.EqualFold(splitId[8], constVirtualMachinesResource) {
+		return "", fmt.Errorf("unexpected format of virtual machine stateful set ID: %v", splitId)
+	}
+	return splitId[7] + "_" + splitId[9], nil
+}
+
+// extract the vmss name from the vmss instance id string
+func getSetForVMSSInstance(in string) string {
+	// Find the index of "virtualMachines" and take only the portion before it
+	trimmed := strings.Split(in, "/"+constVirtualMachinesResource)[0]
+
+	return trimmed
+}
+
+// helper function to return the last part of the interface ID
+func extractResourceSuffix(ifId string) string {
+	tokens := strings.Split(ifId, "/")
+	return tokens[len(tokens)-1]
+}
```

Comment:
Could you add some tests for these string manipulation functions, both so we know they work and to show examples of values that may be expected?